test/base/test_batch.py

```diff
@@ -565,6 +565,47 @@ def test_batch_standard_compatibility() -> None:
         Batch()[0]
 
 
+def test_batch_eq() -> None:
+    # Different keys
```

Comment:
pls add testing equality between non-batch objects

also pls add a test where there's first equality, then a value in an array inside one of the instances is modified which should lead to non-equality.

There's also no coverage of equality being true for simple cases - two empty batches, two batches with the same keys in different orders, two batches where one sequence is passed as tuple and one as list, two batches where a subbatch is passed as dict and the other as `Batch`.

Btw, copilot/ChatGPT can likely generate these tests directly from my comment :)

Reply:
Yep. I also realize that my assert messages are not expressing the opposite of their truthness. For ex.:

```
# Different keys
batch1 = Batch(a=[1, 2], b=[100, 50])
batch2 = Batch(b=[1, 2], c=[100, 50])
assert batch1 != batch2, "Keys are not matching."
```

Should be "Keys should not match." or "Keys are matching."

Reply:
For the future (or now, if you want to): it's better  to have an informative test name and many small tests instead of having assert messages. This way the cases are all executed instead of being aborted after something in the sequence fails. For example, here I would have done:

```python
class TestBatchEquality:

   @staticmethod
   def test_keys_equal():
       assert Batch(a=.., b=...) == Batch(a=.., b=...)
       assert Batch(a=.., b=...) == Batch(b=.., a=...)
    

   @staticmethod
   def test_keys_unequal():
      ...
```

Again, if you asked ChatGPT to refactor this to such a structure, it would likely do a good job

Reply:
Note that now neither comments nor assert messages are necessary